packages/std/src/init_handle.rs

```diff
@@ -143,6 +143,39 @@ where
     }
 }
 
+// some proposed helpers for InitResponse, designed for chaining
+impl<T: Clone + fmt::Debug + PartialEq + JsonSchema> InitResponse<T> {
+    pub fn with_message(mut self, msg: CosmosMsg<T>) -> Self {
+        self.messages.push(msg);
+        self
+    }
+
+    pub fn with_log<U: ToString>(mut self, key: &str, value: U) -> Self {
+        self.log.push(log(key, value));
+        self
+    }
+
+    pub fn with_data<U: Into<Binary>>(mut self, data: U) -> Self {
+        self.data = Some(data.into());
+        self
+    }
```

Comment:
Looking at this more closely, it turns out the interface is not symmetric. The first two appens data to the fields, the last one overrides any existing data.